cosmos/x/evm/plugins/txpool/serializer.go

```diff
@@ -117,7 +117,7 @@ func (s *serializer) SerializeToSdkTx(signedTx *coretypes.Transaction) (sdk.Tx,
 		signingtypes.SignatureV2{
 			Sequence: signedTx.Nonce(),
 			Data: &signingtypes.SingleSignatureData{
-				SignMode: evmante.SignMode_SIGN_MODE_ETHEREUM,
+				SignMode: signingtypes.SignMode(int32(evmante.SignMode_SIGN_MODE_ETHEREUM)), // TODO: this is ghetto af.
```

Comment:
yeah I was trying to find a way around this. Its weird that this SingleSignatureData has the old type for SignMode

Reply:
yeah its a mess since now these int32s are part of a proto